### DISCUSSION TITLE
First time git push with problems

### DISCUSSION THREAD

**Discussion by Wm-M-Thompson** [community/community#34963] [New to GitHub]
I am trying something new.  I am trying to connect  a linux distribution to a new github repository and do a first time push of a helloworld.java file (maybe the problem is that it is a blank file?).  I am using a ubuntu virtual machine but I am sure the issue is not a network connection one.  Here is a screen shot of the issue I am having.  what exactly does "error: src refspec master does not match any" mean?
![033_Screenshot 2022-10-03 074359](https://user-images.githubusercontent.com/5104289/193601935-73c12a1b-8057-4d24-8ac5-3444d0826338.png)

**Comment by airtower-luna** [comment]
You're trying to push a branch called `master`, but your repository doesn't have a branch of that name. The `git status` output shows that you have a branch called `main`, pushing that instead should work. :slightly_smiling_face:

**Reply by Wm-M-Thompson** [reply]
What is the issue and the resolution for using a password?

![034_Screenshot 2022-10-03 074359](https://user-images.githubusercontent.com/5104289/193682980-06a92d1c-15dd-4b1d-b848-a1d96e4cc48a.png)

**Reply by airtower-luna** [reply]
The solution is: Don't use your password, like the message says! :smile_cat:

You cannot use your password for `git push` or other Git operations (but you still need it to log in). You need to [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and enter that instead when Git asks for your password. When creating the token make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).
